servers/media-server/src/server/connector/transports/nats.rs

```diff
@@ -0,0 +1,35 @@
+use super::Transporter;
+
+pub struct NatsTransporter {
+    pub conn: nats::Connection,
+    pub subject: String,
+    pub sub: Option<nats::Subscription>,
+}
+
+impl NatsTransporter {
+    pub fn new(uri: String, subject: String) -> Result<Self, String> {
+        let res = nats::connect(&uri);
```

Comment:
Are you using SYNC version of NATS? It it better to switch to ASYNC version for better performance and avoiding blocking other part.